### ISSUE TITLE
override.typaram.invalid false positive on F-bounded/self-referential type-parameter bounds

### ISSUE THREAD

**Issue by wmdietl** [eisop/checker-framework#1971]
`BaseTypeVisitor.OverrideChecker.isTypeParameterBoundOverrideValid` (added in #1961) compares an overriding and an overridden method's type-parameter bounds structurally, without first substituting the overriding method's type variables for the overridden method's the way JLS 8.4.2 does before checking override compatibility. When a bound mentions a type variable of the same method -- an F-bound such as `<T extends Comparable<T>>`, or one type parameter's bound naming another (`<A extends Object, B extends List<A>>`) -- the comparison ends up comparing the overridden method's own type variable against the overriding method's distinct one, in an invariant type-argument position, and fails regardless of which direction the qualifier moved.

```java
static class Super {
    <T extends @NonNull Comparable<T>> void use(T p) {}
}

static class Sub extends Super {
    @Override
    // Sound: at any instantiation T := C, C <: @NonNull Comparable<C> implies
    // C <: @Nullable Comparable<C>. Rejected anyway.
    <T extends @Nullable Comparable<T>> void use(T p) {}
}
```

This is a precision regression, not a soundness hole -- the check is over-strict, never over-permissive -- so it doesn't block #1961, but it's a real false positive: any override that widens/narrows the qualifier on a self-referential or cross-referencing bound gets `override.typaram.invalid` regardless of direction. Identical bounds on both sides are unaffected (the common `<T extends Comparable<T>>` pattern with no qualifier change compiles clean), so this only surfaces when someone's override actually changes such a bound's qualifier.

Reproduction and a positive/negative control pair (`Super17`/`SubFBoundWiden`, `Super18`/`SubFBoundSame`) are in `checker/tests/nullness/OverrideTypeParamBound.java` on the #1961 branch, along with a `// ---- Known imprecision ----` comment explaining the root cause in more detail.

The fix would be to substitute the overriding method's type variables into the overridden method's bound before comparing, matching JLS 8.4.2's own override-compatibility check. That touches the central comparison in `isTypeParameterBoundOverrideValid`, so it's left for a follow-up rather than folded into #1961.
